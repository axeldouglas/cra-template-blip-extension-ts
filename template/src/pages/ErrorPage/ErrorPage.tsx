interface IErrorPage {
    code?: number;
    text?: string;
}

function ErrorPage({ code = 404, text = 'Page not found' }: IErrorPage) {
    return (
        <div className="flex flex-col justify-center items-center my-24 w-full">
            <h1 className="p-0 m-0 text-9xl text-transparent bg-clip-text bg-gradient-to-r from-primary-purples-acai to-primary-main">
                {code}
            </h1>
            <h2 className="mt-1 text-xl font-normal text-transparent bg-clip-text bg-gradient-to-r from-primary-purples-acai to-primary-main">
                {text}
            </h2>
        </div>
    );
}

export default ErrorPage;
