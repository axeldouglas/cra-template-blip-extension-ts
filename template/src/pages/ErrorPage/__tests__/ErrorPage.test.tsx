import { render } from '@testing-library/react';

import ErrorPage from '../ErrorPage';

describe('<ErrorPage />>', () => {
    it('should render the Error Page', () => {
        const { container } = render(<ErrorPage />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
