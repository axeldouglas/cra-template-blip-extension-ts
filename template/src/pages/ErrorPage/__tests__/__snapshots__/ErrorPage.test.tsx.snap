// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage />> should render the Error Page 1`] = `
<div
  class="flex flex-col justify-center items-center my-24 w-full"
>
  <h1
    class="p-0 m-0 text-9xl text-transparent bg-clip-text bg-gradient-to-r from-primary-purples-acai to-primary-main"
  >
    404
  </h1>
  <h2
    class="mt-1 text-xl font-normal text-transparent bg-clip-text bg-gradient-to-r from-primary-purples-acai to-primary-main"
  >
    Page not found
  </h2>
</div>
`;
